addons/project/models/project_project.py

```diff
@@ -1140,3 +1148,106 @@ def _thread_to_store(self, store: Store, fields, *, request_list=None):
+        if self.collaborator_ids:
+            return [self.env._("Coverting this project to template will clear all the collaborator linked")]
+        return False
+
+    def _is_template_linked(self):
+        return False
+
+    def _get_notification_action(self, tag, params):
+        return {
+            'type': 'ir.actions.client',
+            'tag': tag,
+            'params': params,
+        }
+
+    def action_convert_project_to_template(self):
+        self.ensure_one()
+        if self.is_template:
+            body_message = self._is_template_linked() or {}
+            return self._get_notification_action(
+                "project_template_show_undo_confirmation_dialog",
+                {
+                "project_id": self.id,
+                "message": self._is_template_linked()
+                } | body_message,
+            )
+        if self._is_project_linked():
+            return self._get_notification_action(
+                "project_template_show_confirmation",
+                {
+                    "project_id": self.id,
+                    "next": {
+                        "type": "ir.actions.client",
+                        "tag": "soft_reload",
+                    },
+                    "message": self._is_project_linked()
+                },
+            )
+        return self.convert_project_to_template()
+
+    def convert_project_to_template(self):
+        self.ensure_one()
+        self.collaborator_ids.unlink()
+        self.is_template = True
+        self.message_post(body=self.env._("Project converted to template."))
+        return self._get_notification_action(
+            "project_template_show_notification",
+            {
+                "project_id": self.id,
+                "next": {
+                    "type": "ir.actions.client",
+                    "tag": "soft_reload",
+                },
+            },
+        )
+
+    def action_undo_convert_to_template(self):
+        self.ensure_one()
+        self.is_template = False
+        self.message_post(body=self.env._("Template converted back to regular project."))
+        return self._get_notification_action(
+            "display_notification",
+            {
+                "message": self.env._("Template converted back to regular project."),
+                "next": {
+                    "type": "ir.actions.client",
+                    "tag": "soft_reload",
+                },
+            },
+        )
+
+    @api.model
+    def _get_template_default_context_whitelist(self):
+        """
+        Whitelist of fields that can be set through the `default_` context keys when creating a task from a template.
+        """
+        return []
+
+    @api.model
+    def _get_template_field_blacklist(self):
+        """
+        Blacklist of fields to not copy when creating a task from a template.
+        """
+        return [
+            "partner_id",
+        ]
+
+    def action_create_from_template(self, values={}):
```

Comment:
We should not set a mutable object as default parameter. Would be better to set to None if you need a default value.